ci/pkg/ci/ci.go

```diff
@@ -0,0 +1,327 @@
+		containerField, ok := podSpec[field]
+		if !ok {
+			continue
+		}
+		containers := containerField.([]interface{})
+		for _, container := range containers {
+			images = append(images, container.(map[string]interface{})["image"].(string))
+		}
+	}
+	return images
+}
+
+// GetRepoTags returns the repotags from a tarball of a an image.
+func GetRepoTags(path string) ([]string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	tarReader := tar.NewReader(f)
+	for {
+		header, err := tarReader.Next()
+
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+		if header.Name != "manifest.json" {
+			continue
+		}
+		bytes, err := ioutil.ReadAll(tarReader)
+		if err != nil {
+			return nil, err
+		}
+		jsonBody := make([]interface{}, 0)
+		err = json.Unmarshal(bytes, &jsonBody)
+		if err != nil {
+			return nil, err
+		}
+		allRepoTags := make([]string, 0)
+		for _, imageDef := range jsonBody {
+			repoTags := imageDef.(map[string]interface{})["RepoTags"].([]interface{})
+			for _, tag := range repoTags {
+				allRepoTags = append(allRepoTags, tag.(string))
+			}
+		}
+		return allRepoTags, nil
+	}
+	return nil, nil
+}
+
+// SendResults sends the results to Insights
+func SendResults(reports []ReportInfo, resources []Resource, configurationObject Configuration, token string) error {
+	var b bytes.Buffer
+	w := multipart.NewWriter(&b)
+
+	for _, report := range reports {
+		fw, err := w.CreateFormFile(report.Report, report.Filename)
+		if err != nil {
+			logrus.Warnf("Unable to create form for %s", report.Report)
+			return err
+		}
+		r, err := os.Open(configurationObject.Options.TempFolder + "/" + report.Filename)
+		if err != nil {
+			logrus.Warnf("Unable to open file for %s", report.Report)
+			return err
+		}
+		defer r.Close()
+		_, err = io.Copy(fw, r)
+
+		if err != nil {
+			logrus.Warnf("Unable to write contents for %s", report.Report)
+			return err
+		}
+	}
+	w.Close()
+
+	masterHash, err := GetResultsFromCommand("git", "merge-base", "HEAD", "master")
+	if err != nil {
+		logrus.Warn("Unable to get GIT merge-base")
+		return err
+	}
+
+	currentHash, err := GetResultsFromCommand("git", "rev-parse", "HEAD")
+	if err != nil {
+		logrus.Warn("Unable to get GIT Hash")
+		return err
+	}
+
+	branchName, err := GetResultsFromCommand("git", "rev-parse", "--abbrev-ref", "HEAD")
+	if err != nil {
+		logrus.Warn("Unable to get GIT Branch Name")
+		return err
+	}
+
+	origin, err := GetResultsFromCommand("git", "remote", "get-url", "origin")
+	if err != nil {
+		logrus.Warn("Unable to get GIT Origin")
+		return err
+	}
+
+	if strings.LastIndex(origin, "@") > 3 { // git@github.com URLs are allowed
+		logrus.Warn("Detected possible token in Git Origin, stripping git origin.")
+		originSplit := strings.Split(origin, "@")
+		// Take the substring after the last @ to avoid any tokens in an HTTPS URL
+		origin = originSplit[len(originSplit)-1]
```

Comment:
this will generate inconsistent origin URLs, since those that don't contain an `@` will still have the protocol